service/s3.h

```diff
@@ -261,9 +259,21 @@ struct S3Api : public AwsApi {
         std::string contentEncoding;
         std::map<std::string, std::string> metadata;
         std::string acl;
+    };
+
+    /** A set of options that are used when performing uploads. */
+    struct UploadOptions {
```

Comment:
May be I don't understand what the metadata object is doing but it seems that it does set ACL here:
https://github.com/datacratic/soa/blob/PLAT-996_s3-canned-acl/service/s3.cc#L1450